src/formatDate.js

```diff
@@ -9,6 +9,35 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const spliteOldDate = date.split(oldSeparator);
+
+  const newDate = Array(3);
+
+  for (let i = 0; i < spliteOldDate.length; i++) {
+    const oldPart = fromFormat[i];
+
+    let index = toFormat.indexOf(oldPart);
+
+    newDate[index] = spliteOldDate[i];
+
+  }
+
+  if( toFormat.includes('YY') && !fromFormat.includes('YY')){
+    const index = fromFormat.indexOf('YYYY');
+    newDate[toFormat.indexOf('YY')] = spliteOldDate[index].slice(2);
```

Comment:
The check for `YY` in `toFormat` and its absence in `fromFormat` is correct, but you should ensure that `index` is not `-1` before proceeding to avoid potential errors.